query/releasesearch.go

```diff
@@ -152,16 +159,48 @@ func (s Sort) ESString() string {
 	return esSortNames[s]
 }
 
+type ReleaseType int
+
+const (
+	InvalidReleaseType ReleaseType = iota
+	Upcoming
+	Published
+	Cancelled
+)
+
+var relTypeNames = map[ReleaseType]string{Upcoming: "type-upcoming", Published: "type-published", Cancelled: "type-cancelled", InvalidReleaseType: "Invalid"}
+
+func ParseReleaseType(s string) (ReleaseType, error) {
+	for rt, rtn := range relTypeNames {
+		if strings.EqualFold(s, rtn) {
+			return rt, nil
+		}
+	}
+
+	return InvalidReleaseType, errors.New("invalid release type string")
+}
+
+func MustParseReleaseType(s string) ReleaseType {
+	rt, err := ParseReleaseType(s)
+	if err != nil {
+		panic("invalid release type string: " + s)
```

Comment:
not sure if we have to panic for invalid release type. Can we have a ```log.fatal``` instead please ? so that it will print and exit with code 1.

Reply:
Will add to next PR